source/DurationPicker.js

```diff
@@ -0,0 +1,733 @@
+					this.$.second.set('disabled' , true);
+			} else {
+				if (this.$.hour)
+					this.$.hour.set('disabled', false);
+				if (this.$.minute)
+					this.$.minute.set('disabled', false);
+				if (this.$.second)
+					this.$.second.set('disabled' , false);
+			}
+		},
+
+		/**
+		* @private
+		*/
+		startTapped: function () {
+			if (this.startConent == moon.$L('Start')) {
+				if(this.valueArray['hour'] || this.valueArray['minute'] || this.valueArray['second']) {
+					this.startPicker();
+				}
+			} else {
+				this.set('value', '');
+			}
+		},
+
+		/**
+		* @private
+		*/
+		pauseTapped: function () {
+			if (this.pauseConent == moon.$L('Pause')) {
+				this.set('pauseConent', moon.$L('Resume'));
+				this.pauseTimer();
+			} else {
+				this.set('pauseConent', moon.$L('Pause'));
+				this.startTimer();
+			}
+		},
+
+		/**
+		* @private
+		*/
+		startPicker: function () {
+			this.set('startConent', moon.$L('Cancel'));
+			this.set('pauseDisabled', false);
+			this.startTimer();
+		},
+
+		/**
+		* @private
+		*/
+		cancelPicker: function () {
+			this.set('startConent', moon.$L('Start'));
+			this.set('pauseConent', moon.$L('Pause'));
+			this.set('pauseDisabled', true);
+			this.pauseTimer();
+		},
+
+		/**
+		* @private
+		*/
+		startTimer: function () {
+			window.clearInterval(this.timer);
+			if (this.value) {
+				this.set('countdown', true);
+				this.timer = window.setInterval(
+					enyo.bind (this, function () {
+						this.tick();
+					}), 1000);				
+			}
+		},
+
+		pauseTimer: function () {
+			this.set('countdown', false);
+			window.clearInterval(this.timer);
+		},
+
+		/**
+		* @private
+		*/
+		deletePickers: function () {
+			if (this.$.hourPicker) this.$.hourPicker.destroy();
+			if (this.$.minutePicker) this.$.minutePicker.destroy();
+			if (this.$.secondPicker) this.$.secondPicker.destroy();
+		},
+
+		/**
+		* @private
+		*/
+		resetPicker: function () {
+			if (this.startConent == moon.$L('Cancel')) {
+				this.cancelPicker();
+			}
+			this.valueArray['hour'] = 0;
+			if (this.$.hour) this.$.hour.set('value', 0);
+
+			this.valueArray['minute'] = 0;
+			if (this.$.minute) this.$.minute.set('value', 0);
+
+			this.valueArray['second'] = 0;
+			if (this.$.second) this.$.second.set('value',0);
+		},
+
+		/**
+		* @private
+		*/
+		closePicker: function (sender, ev) {
+			/**
+			* If select/enter is pressed on any date picker item or the left key is pressed on the
+			* first item, close the drawer
+			*/
+			if (ev.type == 'onSpotlightSelect' ||
+				this.$.client.children[0].id == ev.originator.id) {
+				this.expandContract();
+				return true;
+			}
+		},
+
+		/**
+		* @private
+		*/
+		toggleActive: function () {
+			if (this.get('open')) {
+				this.set('active', false);
+				if (!enyo.Spotlight.getPointerMode()) {
+					enyo.Spotlight.spot(this.$.headerWrapper);
+				}
+			} else {
+				this.set('active', true);
+			}
+		},
+
+		/**
+		* @private
+		*/
+		hourTextChanged: function (inOldvalue, inNewValue) {
```

Comment:
Can we update the parameter names so that they do not use the "in" prefix here? Similarly for the other two change handlers that follow.